pennylane/data/__init__.py

```diff
@@ -24,7 +23,175 @@
 .. autosummary::
    :toctree: api
```

Comment:
I'm on the fence, but do we wanna customize this? The [functions/classes section today](https://xanaduai-pennylane--4097.com.readthedocs.build/en/4097/code/qml_data.html#functions) includes every `DatasetAttribute` type, and the `field()` function, but those aren't really intended for general use as far as I'm aware. thoughts? I'm happy to leave them there, but also happy to remove them if we actively want users to avoid learning about the internals

Reply:
Sure, but perhaps we can do this later? Personally I'm okay with users being able to explore the internals if they want to.